### DISCUSSION TITLE
Incorrect Company Values - Are These Gameplay Changes Or Defect Fixes?

### DISCUSSION THREAD

**Discussion by benda** [OpenTTD/OpenTTD#9765] [Features]
I didn't see this discussed anywhere else.

I am curious if these changes would be considered game play changes and thus any pull requests would not be accepted? In my opinion these are defect fixes and not gameplay changes, but of course the opinion of the project maintainers is what matters.

1. Company value calculation is missing the value of the company's shares in other companies. This can be a huge number, and is what causes huge drops in company values as seen on the company value graph when you buy shares in other companies. That drop is inaccurate. Due to the way company values are calculated currently, the value of shares in other companies is unclear, since those other companies may themselves hold shares of other companies. A simple yet not entirely accurate way would be just to get all of the company values without including value of shares in other companies, then add in the value of the shares which now is known. See screenshot below.

2. The current company value calculation just sums the value of all assets, but the values assigned to assets is unclear / arbitrary. The value of the assets can be more accurately determined by looking at real world public markets (using CSX as an example, it has $33 billion in Property, Plant & Equipment assets (i.e. stations & vehicles) and an $80 billion market value, so that would be an asset multiple of 2.4 - which also means real world public markets don't value railroad company's assets very highly).

3. Valuing a company based on value of assets is not generally the right way to value a company. A more commonly used method would be to take the last year's net income of the company multiplied by some standard multiplier, that can be determined from real world public markets. For CSX, it has $3.6 billion in net income and $80 billion market value, so the  multiplier is 22. You don't need to deal with depreciation or liabilities of the company or any of that; in theory the net income is already doing that. This also means the value of all the company's assets is the net income that the company was able to derive from those assets. 

https://seekingalpha.com/symbol/CSX/income-statement
https://seekingalpha.com/symbol/CSX/balance-sheet

Example of Issue 1 
My company value isn't $2,710,980 when I own 75% of a company valued at $34,912,508. It should be, at least with simplistic but relatively accurate mechanics, $2,710,980 + (.75 *  $34,912,508) = $28,895,361. That is a much more accurate number than the original $2,710,980.

The drops in my company's value on the graph are when I bought the shares. Those drops don't make sense; in reality the value of my company has not changed at all by buying the shares.

![image](https://user-images.githubusercontent.com/591816/147271809-def9f208-b79b-4ef5-8e8f-cf7f562a0d38.png)

This is how it looks fixed in my fork, and the cool thing here is that my company value now keeps pace with the companies I own, as it should.
![image](https://user-images.githubusercontent.com/591816/147278009-5332f54a-9f8d-442d-b7b2-7f1eaba5d921.png)

**Comment by nielsmh** [comment]
In my eyes it would be a gameplay change, but that doesn't mean it would be rejected automatically. Rather, it means that it needs to demonstrate it's a change for the better all in all, e.g. that it allows new or improved gameplay, while not being in the way of existing gameplay.

**Comment by benda** [comment]
1. I am able to code this myself and submit a PR. 
2. How specifically can I "demonstrate it's a change for the better all in all, e.g. that it allows new or improved gameplay, while not being in the way of existing gameplay." ? I'm happy to do whatever you think is necessary here.

**Comment by rubidium42** [comment]
The problem with accounting for shares in company value is that company A can own shares in company B and vice versa. This gives an infinite dependency loop for the company value.

**Reply by benda** [reply]
I addressed this in my original post - "A simple yet not entirely accurate way would be just to get all of the company values without including value of shares in other companies, then add in the value of the shares which now is known." See screenshot & explanation by it for an example.

**Comment by LC-Zorg** [comment]
It's nice that you got interested in this topic. In my opinion, there is a more than one thing to improve here. There is lot to discuss and determine what direction of change might be good for the game. :)

Imho, firstly, I believe that the valuation of the company's value using the property method is highly flawed in this game - the money you have rarely shows how much the company is currently worth. I'm of the opinion that a much better and more reliable valuation would be the income method, i.e. on the basis of how much the company is able to bring in profit in a given period of time. It could also be a combined method considering the value of infrastructure and vehicles.
Changing the method would also remove a significant disadvantage of the current method of buyout of companies, where you pay for the company's assets, which are finally taken over, so the actual cost of taking over any, even a powerful company, is in most situations close to zero.

More than a year ago, there was a [discussion](https://www.tt-forums.net/viewtopic.php?t=87137) on the tt-forums about changes to the game economy (here). I added a few of [my comments and suggestions](https://www.tt-forums.net/viewtopic.php?p=1232660#p1232660) there, also regarding the method of valuation of the company - point 8 and regarding the sale and value of shares - point 3

**Reply by nielsmh** [reply]
If you bring up taking over companies/buying shares (and I understand why, apart from scoring purposes it's the only thing the company value is used for), then you have to consider a lot more if you want to improve it and make stock ownership make sense. You need to implement a more realistic ownership model, where stock can be bought and sold more fine grained than 25% of total, and the game would need to distinguish between shares owned by "the public" and shares owned by the company.
I remember playing Railroad Tycoon (the 1991 Deluxe version) where your company starts out owned 100% by the public, but you can buy _treasury stock_, i.e. buy back the stock from the public to lessen the influence shareholders can have and prevent hostile takeovers. Occasionally the stock splits into more shares, and occasionally you get an offer to sell new stock, where you get a lump sum of money (investment) but a number of new shares are generated that are owned by the public.
And in Railroad Tycoon 2, you play as a managing director (or similar) who's a distinct entity from the company or companies you control. You have a personal fortune and can have personal ownership of part of whole of companies.

A big issue with implementing all this is just the usual with it being far too easy to earn money, eventually you're swimming in cash and can just buy 100% of your own stock and be immune to anything, won't even have to pay dividends. A ton of things will need to be changed/"fixed" to make this viable gameplay.

**Reply by benda** [reply]
I know that for me personally, getting enough money to take over all your competitors was how to "win" the game. So I always found this portion of the game to be very important. If companies were valued as LC-Zorg and I are recommending, it would cost a lot more to take over companies so the fact that it is relatively easily to get money isn't an issue since you need more of it.
